### PR TITLE
Update Github demo to use beta2.

### PR DESCRIPTION
**Pull request by yigit**
This is a slightly bigger change due to espresso version update and
removal of alpha APIs in arch components.

### CODE REVIEW

File: GithubBrowserSample/app/build.gradle [hunk 1]
```diff
@@ -104,7 +106,8 @@ dependencies {
 
     androidTestImplementation("android.arch.core:core-testing:$arch_version", {
     })
-    androidTestImplementation "org.mockito:mockito-android:$mockito_version"
+    androidTestImplementation "com.linkedin.dexmaker:dexmaker-mockito:$dexmaker_version", { exclude group: 'net.bytebuddy' }
```
Comment (florina-muntenescu) [inline GithubBrowserSample/app/build.gradle]: why is this needed?

File: GithubBrowserSample/app/src/main/java/com/android/example/github/MainActivity.java [hunk 2]
```diff
@@ -18,32 +18,31 @@
...
+import android.arch.lifecycle.LifecycleObserver;
+import android.arch.lifecycle.LifecycleOwner;
 import android.arch.lifecycle.LifecycleRegistry;
 import android.arch.lifecycle.LifecycleRegistryOwner;
+import android.arch.lifecycle.OnLifecycleEvent;
 import android.os.Bundle;
 import android.support.annotation.Nullable;
 import android.support.v4.app.Fragment;
 import android.support.v7.app.AppCompatActivity;
+import android.util.Log;
```
Comment (florina-muntenescu) [inline GithubBrowserSample/app/src/main/java/com/android/example/github/MainActivity.java]: import doesn't seem to be used.

File: GithubBrowserSample/app/src/main/java/com/android/example/github/ui/user/UserFragment.java [hunk 3]
```diff
@@ -46,7 +46,7 @@
 
     DataBindingComponent dataBindingComponent = new FragmentDataBindingComponent(this);
     private UserViewModel userViewModel;
-    private AutoClearedValue<UserFragmentBinding> binding;
+    AutoClearedValue<UserFragmentBinding> binding;
```
Comment (florina-muntenescu) [inline GithubBrowserSample/app/src/main/java/com/android/example/github/ui/user/UserFragment.java]: why not private?
If it's for testing, can you add @VisibleForTesting?

File: GithubBrowserSample/app/build.gradle (hunk 1 above)
Comment (yigit) [reply GithubBrowserSample/app/build.gradle]: the arch testing incremented version in between that comes w/ this byte buddy problem.
(see build files of arch components for details). I think it is fixed in later versions but didn't want to try that.

File: GithubBrowserSample/app/src/main/java/com/android/example/github/MainActivity.java (hunk 2 above)
Comment (yigit) [reply GithubBrowserSample/app/src/main/java/com/android/example/github/MainActivity.java]: removed.
